File: snippets/node/esearch.js
let searchParams = new kaltura.objects.ESearchEntryParams();
searchParams.searchOperator = new kaltura.objects.ESearchEntryOperator();
searchParams.searchOperator.searchItems = [];
searchParams.searchOperator.searchItems[0] = new kaltura.objects.ESearchUnifiedItem();
searchParams.searchOperator.searchItems[0].searchTerm = "kaltura logo";
searchParams.searchOperator.searchItems[0].itemType = kaltura.enums.ESearchItemType.EXACT_MATCH;
searchParams.searchOperator.searchItems[0].addHighlight = true;
let pager = new kaltura.objects.Pager();

kaltura.services.eSearch.searchEntry(searchParams, pager)
.execute(client)
.then(result => {
    console.log(result);
});

File: snippets/java/esearch.java
// 1 Params and Operator 
ESearchEntryParams searchParams = new ESearchEntryParams();
searchParams.setSearchOperator(new ESearchEntryOperator());
searchParams.getSearchOperator().setSearchItems(new ArrayList<ESearchEntryBaseItem>(1));

// 2 Search Type 
searchParams.getSearchOperator().getSearchItems().set(0, new ESearchUnifiedItem());

// 3 Search Term
searchParams.getSearchOperator().getSearchItems().get(0).setSearchTerm("kaltura logo");

// 4 Search Item Type 
searchParams.getSearchOperator().getSearchItems().get(0).setItemType(ESearchItemType.EXACT_MATCH.getValue());

// 5 Add Highlight 
searchParams.getSearchOperator().getSearchItems().get(0).setAddHighlight(true);

// 6 Search 
SearchEntryESearchBuilder requestBuilder = ESearchService.searchEntry(searchParams)
    .setCompletion(new OnCompletion<Response<ESearchEntryResponse>>() {
        @Override
        public void onComplete(Response<ESearchEntryResponse> result) {
            System.out.println(result);
        }
    });